php/libraries/Utility.class.inc

```diff
@@ -558,7 +558,7 @@ class Utility
      */
     static function asArray($var)
     {
-        if (!is_array($var)) {
+        if (!is_array($var) && !empty($var)) {
```

Comment:
Another PR can propose changing it, I just don't think it should be done so close to a release because of how far reaching the change is in terms of how low level it is and how many modules and instruments that use the function.